src/utransport.rs

```diff
@@ -230,38 +240,47 @@ pub trait UTransport: Send + Sync {
     /// #     }
     /// # }
     /// #
-    /// # let my_transport = MyTransport::new();
-    /// # let my_uuri = UUri::default();
+    /// # let mut my_transport = MyTransport::new();
+    /// # let source_uuri = UUri::default();
     ///
     /// // hang onto this listener...
-    /// let my_listener: Arc<dyn UListener> = Arc::new(MyListener::new());
-    /// // ...send a clone through when registering...
-    /// let register_result = my_transport.register_listener(my_uuri.clone(), my_listener.clone());
+    /// let my_listener = Arc::new(MyListener::new());
+    /// // ...use a clone for registering...
+    /// my_transport.register_listener(
+    ///     Some(&source_uuri),
+    ///     None,
+    ///     my_listener.clone());
     /// // ...and use the original we hung onto when unregistering
-    /// let unregister_result = my_transport.unregister_listener(my_uuri, my_listener);
+    /// my_transport.unregister_listener(
+    ///     Some(&source_uuri),
+    ///     None,
+    ///     my_listener);
     /// ```
     async fn register_listener(
-        &self,
-        topic: UUri,
+        &mut self,
+        source_filter: Option<&UUri>,
+        sink_filter: Option<&UUri>,
         listener: Arc<dyn UListener>,
     ) -> Result<(), UStatus>;
 
-    /// Unregisters a listener for a given topic.
+    /// Unregisters a listener.
     ///
-    /// Messages arriving on this topic will no longer be processed by this listener.
+    /// The listener will no longer be called for any (matching) messages after this function has
+    /// returned successfully.
     ///
     /// # Arguments
     ///
-    /// * `topic` - Resolved topic uri where the listener was registered originally.
-    /// * `listener` - Identifier of the listener that should be unregistered. Here we take ownership
-    ///                to communicate that this listener is now finished.
+    /// * `source` - The source address pattern that the listener had been registered for.
+    /// * `sink` - The sink address pattern that the listener had been registered for.
+    /// * `listener` - The listener to unregister.
     ///
     /// # Errors
     ///
     /// Returns an error if the listener could not be unregistered, for example if the given listener does not exist.
     async fn unregister_listener(
-        &self,
-        topic: UUri,
+        &mut self,
```

Comment:
Same question as up [here](https://github.com/eclipse-uprotocol/up-rust/pull/92/files#r1583558975), but:

Why is an exclusive reference needed instead of a shared reference in the new formulation?